src/fosslight_reuse/_add.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021 LG Electronics Inc.
+# SPDX-License-Identifier: GPL-3.0-only
+import os
+import re
+import logging
+import fosslight_util.constant as constant
+from yaml import safe_dump
+from fosslight_util.set_log import init_log
+from datetime import datetime
+from ._fosslight_reuse import reuse_for_project, reuse_for_files, print_error
+from reuse.header import run
+from reuse._comment import EXTENSION_COMMENT_STYLE_MAP_LOWERCASE
+from reuse._main import parser as reuse_arg_parser
+
+_PKG_NAME = "fosslight_reuse"
+_check_only_file_mode = False
+_auto_add_mode = False
+_result_log = {}
+logger = logging.getLogger(constant.LOGGER_NAME)
+
+
+def check_file_extension(file_list):
+    files_filtered = []
+
+    if file_list != "":
+        for file in file_list:
+            file_extension = os.path.splitext(file)[1]
+            if file_extension in EXTENSION_COMMENT_STYLE_MAP_LOWERCASE:
+                files_filtered.append(file)
+
+    return files_filtered
+
+
+def check_license_and_copyright(path_to_find, all_files, missing_license, missing_copyright):
+    # Check file extension for each list
+    all_files_fitered = check_file_extension(all_files)
+    missing_license_filtered = check_file_extension(missing_license)
+    missing_copyright_filtered = check_file_extension(missing_copyright)
+
+    skip_files = sorted(list(set(all_files_fitered) - set(missing_license_filtered) - set(missing_copyright_filtered)))
+    logger.info("# File list that have both license and copyright : {count} / {total}".format(
+            count=len(skip_files),
+            total=len(all_files)))
+
+    for file in skip_files:
+        file_list = list()
+        file_list.append(file)
+
+        unused_lic_list, usused_cop_list, error_occurred, project = reuse_for_files(path_to_find, file_list)
+
+    return missing_license_filtered, missing_copyright_filtered
+
+
+def convert_to_spdx_format(input_string):
+    input_string = input_string.replace(" ", "-")
```

Comment:
Make it work only if input string is not SPDX-ID.

Reply:
Okay I'll add code to check if the input string is SPDX-ID

Reply:
Let's add the function for checking SPDX next time...